runtime/src/in_mem_accounts_index.rs

```diff
@@ -643,8 +684,8 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
 
         // load from disk
         if let Some(disk) = self.bucket.as_ref() {
-            let items = disk.items_in_range(range);
             let mut map = self.map().write().unwrap();
+            let items = disk.items_in_range(range); // map's lock has to be held while we are getting items from disk
```

Comment:
I wonder if there's a way to enforce this ordering requirement in code. Like if the fn took a MutexGuard as a parameter, or if the `items_in_range()` was also a fn on the `map` (as the MutexGuard).

Not sure how far to take this one either, but correct locking/safety is pretty high on my list. Maybe a test could be added to enforce this requirement? Wouldn't need to be in this PR though.

Reply:
This is something to consider. This is the only caller to disk.items_in_range and the intersection of the 2 locks occurs only in this file. So, these ARE the functions that have to get it right.